doc/specs/vulkan/chapters/memory.txt

```diff
@@ -552,58 +552,57 @@ There must: be at least one memory type with the
 ename:VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT bit set in its
 pname:propertyFlags.
 
-The memory types are sorted according to a preorder which serves to aid in
-easily selecting an appropriate memory type.
-Given two memory types X and Y, the preorder defines [eq]#X {leq} Y# if:
-
-  * the memory property bits set for X are a strict subset of the memory
-    property bits set for Y.
-    Or,
-  * the memory property bits set for X are the same as the memory property
-    bits set for Y, and X uses a memory heap with greater or equal
-    performance (as determined in an implementation-specific manner).
-
-Memory types are ordered in the list such that X is assigned a lesser
-pname:memoryTypeIndex than Y if [eq]#(X {leq} Y) {land} {lnot} (Y {leq} X)#
-according to the preorder.
-Note that the list of all allowed memory property flag combinations above
-satisfies this preorder, but other orders would as well.
-The goal of this ordering is to enable applications to use a simple search
-loop in selecting the proper memory type, along the lines of:
+The array pname:memoryTypes must: be ordered in such a way that for each of
+its distinct memory types X and Y it holds that X occupies a lower index
+position than Y if:
```

Comment:
Perhaps it would make more sense to say this as:

```
For any two array elements *X* and *Y* returned in pname:memoryTypes, *X* must: be at a lower index than *Y* if:
```

Mostly it's just shorter, which usually aids readability. (Though your paragraph is still way better than the previous one IMO)

Reply:
I suppose that makes sense. "be at index" feels informal.

> (Though your paragraph is still way better than the previous one IMO)

That was the initial point. Then it got out of hand...

Reply:
> "be at index" feels informal.

Yea true, there's similar wording elsewhere in the spec that's equally awkward, not sure what a better wording would be though...? Well, maybe:

```
...be placed at a lower index within pname:memoryTypes than *Y*...
```

Not sure if that's better but maybe?